manifests/init.pp

```diff
@@ -54,27 +51,26 @@
 # @author Ger Apeldoorn <info@gerapeldoorn.nl>
 #
 class rhsm (
-  $rh_user                           = undef,
-  $rh_password                       = undef,
-  $org                               = undef,
-  $activationkey                     = undef,
-  $pool                              = undef,
-  $proxy_hostname                    = undef,
-  $proxy_port                        = undef,
-  $proxy_user                        = undef,
-  $proxy_password                    = undef,
-  $baseurl                           = 'https://cdn.redhat.com',
-  $servername                        = 'subscription.rhsm.redhat.com',
-  $serverprefix                      = '/subscription',
-  $serverport                        = 443,
-  $ca_cert_dir                       = '/etc/rhsm/ca/',
-  $repo_ca_cert_filename             = 'redhat-uep.pem',
-  $repo_ca_cert_source               = undef,
-  $manage_repos                      = 1,
-  $full_refresh_on_yum               = 0,
-  $package_ensure                    = 'latest',
-  Array[String[1]] $enabled_repo_ids = [],
-) {
+  Optional[String[1]]    $rh_user               = undef,
```

Comment:
Just FYI: back in the days we explicitly didn't use data types potentially stay compatible with the old puppet provided by Redhat with Satellite 6.

Reply:
As the type/provider code does not work with Puppet 3 Agents, this is fine for Satellite >= 6.4 (cornercase: Puppet 4/5 Server and Puppet 3 Agent)